### PR TITLE
CI: Add the "Doctests" workflow to run doctests weekly

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Add a new workflow "ci_doctests.yaml" to run all doctests. This workflow is scheduled to run weekly on Sunday.

Fixes https://github.com/GenericMappingTools/pygmt/issues/2286.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: .github/workflows/ci_doctests.yaml
Comment (seisman) [inline .github/workflows/ci_doctests.yaml:8]: Comment this line before merging:
```suggestion
  # pull_request:
```

File: doc/maintenance.md
Comment (seisman) [inline doc/maintenance.md:79]: **Off-topic question**: Currently, we're documenting what a workflow does in the maintainers' guide, but it's very likely that we may forget to update these descriptions when we update a workflow. Perhaps we should only keep a short, one-sentence description for each workflow here and have more detailed descriptions in the workflow file.

Comment (weiji14) [reply doc/maintenance.md:79]: Maybe have a README.md under https://github.com/GenericMappingTools/pygmt/tree/v0.9.0/.github/workflows to document, so that people looking at the folder can see what the *.yml files are for? Also fine with including it in the YAML file since that will be more visible.

File: doc/maintenance.md
Comment (weiji14) [inline doc/maintenance.md:97]:
```suggestion
    This workflow is scheduled to run all doctests every Sunday on the *main*
```

File: .github/workflows/ci_doctests.yaml
Comment (weiji14) [inline .github/workflows/ci_doctests.yaml:27]: Do we need to do tests on both Python 3.9 and 3.11? Or can we just do one test (e.g. on Python 3.11)?

File: doc/maintenance.md
Comment (seisman) [reply doc/maintenance.md:79]: > Also fine with including it in the YAML file since that will be more visible.

I'm inclined to this option because (1) we won't forget to update the workflow descriptions; (2) the workflows are more self-explanatory.

File: .github/workflows/ci_doctests.yaml
Comment (seisman) [reply .github/workflows/ci_doctests.yaml:27]: I'm OK with one Python version.

File: doc/maintenance.md
Comment (weiji14) [reply doc/maintenance.md:79]: > > Also fine with including it in the YAML file since that will be more visible.
> 
> I'm inclined to this option because (1) we won't forget to update the workflow descriptions; (2) the workflows are more self-explanatory.

Ok, so just document things in the YAML files and remove the descriptions from the doc/maintenance.md file. Can do this in another PR once this one is merged.

**Review by weiji14 — APPROVED** [review]
Almost ready I think, just need to comment the `pull_request:` line and fix the typo.

File: .github/workflows/ci_doctests.yaml
Comment (seisman) [inline .github/workflows/ci_doctests.yaml:57]:
```suggestion
          mamba install gmt=6.4.0 numpy \
```

File: doc/maintenance.md
Comment (michaelgrund) [inline doc/maintenance.md:136]:
```suggestion
    merged into the *main* branch.
```

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**
